### ISSUE TITLE
object.save doesn't work properly for non-default phoneNumberField

### ISSUE THREAD

**Issue by jdkanani** [Dashride/mongoose-intl-phone-number#1]
Example plugin code:

```
UserSchema.plugin(phoneNumberPlugin, {
    hook: 'validate',
    phoneNumberField: 'phone_number',
    nationalFormatField: 'national_format',
    countryCodeField: 'country_code',
});
```
`user.save()` (while updating `user` object) doesn't set `national_format` and `country_code`.

**Probable reason:**
`this.isDirectModified('phoneNumber')` should be the reason as it uses hard-code default path. Check here: https://github.com/Dashride/mongoose-intl-phone-number/blob/master/lib/mongoose-intl-phone-number.js#L79

**Solution (just a theory, not tested):**
Replacing `this.isDirectModified('phoneNumber')` with `this.isDirectModified(phoneNumberField)` should work.

**Comment by jrthib** [comment]
Ah thanks for this. I will get it updated!

**Comment by jdkanani** [comment]
:+1: Thanks.

**Comment by jrthib** [comment]
The fix is available in 0.4.0
